src/utils/componentEnhancers.jsx

```diff
@@ -0,0 +1,15 @@
+import GenericErrorBoundary from '../components/genericErrorBoundary';
+
+const wrapSectionComponentsWithErrorBoundaries = (sections = []) => {
+  const newSections = [];
```

Comment:
This could be done with a map:

```jsx
return sections.map(section => ({
  ...section,
  rows: section.rows.map(row => row.map((component, index) => (
    <GenericErrorBoundary key={index}>{component}</GenericErrorBoundary>
  ))
}))
```